### PR TITLE
Enable multiple PCI segments on AArch64

### PR DESCRIPTION
**Pull request by michael2012z**
Fixes https://github.com/cloud-hypervisor/cloud-hypervisor/issues/3296

The PR enabled multiple PCI segments support on ACPI only. 
The enabling on FDT depends on some modifications in this PR, will come soon.

### CODE REVIEW

**Comment by michael2012z** [comment]
The added devices look quite normal, but in fact they are not functioning.

Take `virtio-net` test for example:
- `lspci` showed the device on the correct segment;
- `ip a` showed a new interface in "UP" state, with IP address;
- There was not any error message in guest `dmesg`;
- But the VM just cannot be reached.

`virtio-pmem` and `virtio-fs` were similar.

**Comment by rbradford** [comment]
> The added devices look quite normal, but in fact they are not functioning.

Can you add `pci=nocrs` to your kernel parameters for your local testing to see if improves things.

**Comment by michael2012z** [comment]
Unfortunately `pci=nocrs` is not available on ARM64: 
"[    0.000000] PCI: Unknown option `nocrs'".

More observation on virtio-pmem: I tried to create a partition table on the hotplug pmem device with `fdisk /dev/pmem0`. While writing the partition table, the `fdisk` program pended at `fsync` system call. There was not any error in guest kernel messages or in VMM log.

**Comment by michael2012z** [comment]
Update:

The problem turned out related to interrupt handling. Quite probably we have something missed around GIT and ITS.

Specifically:
- I used a hot-plugged virtio-pmem to test. A `fsync` syscall pended on writing the new partition table.
- Debugging the guest kernel, I found the the endless waiting happened here: https://github.com/torvalds/linux/blob/v5.14/drivers/nvdimm/nd_virtio.c#L92
- This https://github.com/torvalds/linux/blob/v5.14/drivers/nvdimm/nd_virtio.c#L13 should be triggered by an interrupt to end the waiting.

So, the expected interrupt never came. Still investigating the root cause.

File: vmm/src/interrupt.rs [hunk 1]
```diff
@@ -353,8 +354,26 @@ pub mod kvm {
                 kvm_route.u.msi.data = cfg.data;
 
                 if vm.check_extension(hypervisor::Cap::MsiDevid) {
+                    // On AArch64, there is limitation on the range of the 'devid',
```
Comment (michael2012z) [inline vmm/src/interrupt.rs:357]: The limitation I mentioned in here: https://github.com/torvalds/linux/blob/v5.14/arch/arm64/kvm/vgic/vgic-its.c#L917

Comment (michael2012z) [reply vmm/src/interrupt.rs:357]: @rbradford @sboeuf What do you think about this part?

**Review by MrXinWang — APPROVED** [review]
Approve for the AArch64 related changes.

File: vmm/src/interrupt.rs (hunk 1 above)
Comment (rbradford) [reply vmm/src/interrupt.rs:357]: I think it's a great solution!

**Review by rbradford — APPROVED** [review]

**Pull request merged**